var/spack/repos/builtin/packages/octa/package.py

```diff
@@ -20,44 +22,131 @@ class Octa(AutotoolsPackage):
         extension="tar.gz",
     )
 
+    variant("withgui", default=True, description="Enables the GUI for GOURMET.")
+    variant(
+        'engine', default='cognac, pasta, sushi, muffin',
+        values=('cognac', 'pasta', 'sushi', 'muffin'),
+        multi=True,
+        description='Simulation programs'
+    )
+
     depends_on("autoconf", type="build")
     depends_on("automake", type="build")
     depends_on("libtool", type="build")
-    depends_on("libjpeg", type="link")
-    depends_on("libpng", type="link")
-    depends_on("zlib-api", type="link")
-    depends_on("jogl")
-    depends_on("python")
-    depends_on("gnuplot", type="run")
-    depends_on("py-numpy")
-    depends_on("py-scipy")
-    depends_on("py-numba")
+    depends_on("mpi")
 
+    with when("+withgui"):
+        depends_on("libjpeg", type="link")
+        depends_on("libpng", type="link")
+        depends_on("zlib", type="link")
+        depends_on("jogl")
+        depends_on("python")
+        depends_on("gnuplot", type="run")
+        depends_on("py-numpy")
+        depends_on("py-scipy")
+        depends_on("py-numba")
+        
     # specify for linux_aarch64
-    patch("aarch64.patch", when="target=aarch64:")
+    #  patch("aarch64.patch", when="target=aarch64:")
+    patch("fj_sushi.patch", when="%fj")
+    patch("aarch64gcc_sushi.patch", when="target=aarch64: %gcc")
+    patch("fj_pasta.patch", when="%fj")
+    patch("aarch64gcc_pasta.patch", when="target=aarch64: %gcc")
+    patch("fj_muffin.patch", when="%fj")
+    patch("fj_cognac.patch", when="%fj")
+    #  patch("gourmeterm.patch", when="target=aarch64:")
```

Comment:
Can you add a brief comment above each `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

Reply:
Also, did you contribute the patches upstream? If so, fetching from a URL would be welcome.

Reply:
@alalazo  I apologize for the late response. 
I have added comments above each patch. I have not yet contributed the patches upstream.